pool-payout-zkapp/src/PoolPayout.ts

```diff
@@ -116,33 +146,48 @@ export class PoolPayout extends SmartContract {
     let validatorPublicKey = this.validatorPublicKey.get();
     this.validatorPublicKey.assertEquals(validatorPublicKey);
 
-    // Assert the epoch is correct
-    epoch.assertEquals(currentEpoch, "The epoch must match");
-
-    var signedData: Field[] = [];
+    let signedData: Field[] = [];
 
-    for (let [_, value] of Object.entries(accounts)) {
-
-      // reconstruct the signed data
-      signedData = signedData.concat(value.index.toFields()).concat(value.publicKey.toFields()).concat(value.rewards.toFields());
+    // starting with the index on state, we can increment this variable during this transaction
+    let transactionIndex = currentIndex;
 
+    for (let i = 0; i < accounts.rewards.length; i++) {
+      const reward = accounts.rewards[i];
+      const accountIsNotEmpty = Bool.not(reward.publicKey.isEmpty());
```

Comment:
This `isEmpty` stuff is based on the idea that not every delegate will have a clean 8-divisible number of delegators.  I'm not sure how you handle this on the graphql side, but here I assumed an empty `Reward` has an empty publicKey,  This assumption is almost guaranteed to be wrong, so you can adjust the implementation details in this line to fit the truth.